### PR TITLE
PCVL-435 shots user documentation

### PR DESCRIPTION
**Pull request by ericbrts**

### CODE REVIEW

File: docs/source/legacy.rst [hunk 1]
```diff
@@ -64,6 +64,29 @@ To version 0.10
 
 .. note:: StateVector will interact badly with any :code:`numpy` scalar type
 
+Shots in algorithms
+^^^^^^^^^^^^^^^^^^^
+
+When instanciating an algorithm class (:code:`Sampler`, :code:`Analyzer`) with a :code:`RemoteProcessor`, the user now has to
+pass a positive integer value for the named parameter :code:`max_shots_per_call`. Please note that this parameter
+name *has to* be typed in order to avoid potential signature errors.
```
Comment (MarionQuandela) [inline docs/source/legacy.rst:72]: double * (**has to**) to put in bold no ?

File: docs/source/legacy.rst [hunk 2]
```diff
@@ -64,6 +64,29 @@ To version 0.10
...
+
+>>> p = RemoteProcessor("sim:platform")
+>>> sampler = Sampler(p, max_shots_per_call=10_000_000)
+
+This parameter is also handled by local simulations.
+
+.. note:: Probability amplitude back-ends used for sampling (e.g. using **SLOS** for a :code:`sample_count` call) cannot
+    estimate accurately the sample to shots ratio when converting probabilities to samples.
+
+:code:`samples` and :code:`sample_count` parameter was renamed from :code:`count` to :code:`max_samples`
```
Comment (MarionQuandela) [inline docs/source/legacy.rst:82]: I don't really understand what has been renamed in what ^^'

File: docs/source/notebooks/Remote_computing.ipynb [hunk 3]
```diff
@@ -2133,20 +2154,33 @@
     }
    },
    "source": [
-    "We can now use the `Sampler` with our `RemoteProcessor`."
+    "We can now use the `Sampler` with our `RemoteProcessor`. You have to set a maximum shots threshold (`max_shots_per_call` named parameter) when creating a `Sampler` with a remote platform. Local simulations do not require this threshold.\n",
+    "A shot is any detected event containing at least one photon, it is easy to explain, easy to measure. This shot threshold will prevent the user from consuming too many QPU resources, as once it's reached, the acquisition stops. Shots up to this threshold can be reached for all jobs generated by `Sampler` calls (e.g. calling `sample_count` thrice can lead to the use of at most `3*max_shots_per_call` shots)."
```
Comment (MarionQuandela) [inline docs/source/notebooks/Remote_computing.ipynb:2158]: This should be somewhere in a .rst

File: docs/source/notebooks/Remote_computing.ipynb [hunk 4]
```diff
@@ -2250,37 +2284,64 @@
     }
    },
    "source": [
-    "You can run the same sampling on the corresponding QPU:"
+    "You can run the same sampling on the corresponding QPU. In order to manage your QPU credits, you can estimate the number of shots you'd need for a particular data acquisition. Please note that the maximum shots and maximum samples number act as a dual threshold system. As soon as one of these thresholds is exceeded, the acquisition stops and the results are returned."
```
Comment (MarionQuandela) [inline docs/source/notebooks/Remote_computing.ipynb:2287]: Same, this should go in a .rst

**Review by MarionQuandela — COMMENTED** [review]
I don't see something about what happens if you specify max_shots and max_samples (like the sim/qpu stop at the first number achieved)

File: docs/source/legacy.rst (hunk 2 above)
Comment (ericbrts) [reply docs/source/legacy.rst:82]: The code example below doesn't explain it well enough?

Comment (MarionQuandela) [reply docs/source/legacy.rst:82]: Do you mean sample_count is also a method ? If so consider putting the sentence "Parameter :code:`count` was renamed :code:`max_samples` in methods :code:`samples` and :code:`sample_count`"

File: docs/source/notebooks/Remote_computing.ipynb (hunk 3 above)
Comment (ericbrts) [reply docs/source/notebooks/Remote_computing.ipynb:2158]: Reworked algorithms.rst a lot

File: docs/source/notebooks/Remote_computing.ipynb (hunk 4 above)
Comment (ericbrts) [reply docs/source/notebooks/Remote_computing.ipynb:2287]: Reworked algorithms.rst a lot

File: docs/source/algorithms.rst [hunk 5]
```diff
@@ -74,59 +68,78 @@ complex preconfigured processor from a gate-based circuit.
...
+Cloud with rights is required to follow this tutorial: :ref:`Remote computing on Quandela Cloud`).
 
-A given processor is only able to support some specific types of commands related to its hardware implementation. For
-instance, a real QPU is natively able to sample output detections, but not to compute probabilities of output states
+Once you have created a token suiting your needs (it needs to be given the rights to run jobs on target platforms), you
+may save it once and for all on your computer by running:
+
+>>> pcvl.save_token('YOUR_TOKEN')
+
+.. note:: We recommend you save your token only on a personal computer, not on shared/public ones.
```
Comment (MarionQuandela) [inline docs/source/algorithms.rst:81]: 👍

**Review by MarionQuandela — APPROVED** [review]

**Pull request merged**